src/libs/Middleware/Reauthentication.js

```diff
@@ -113,6 +111,15 @@ function Reauthentication(response, request, isFromSequentialQueue) {
 
             // Return response data so we can chain the response with the following middlewares.
             return data;
+        })
+        .catch((error) => {
+            // If the request is on the sequential queue, re-throw the error so we can decide to retry or not
+            if (isFromSequentialQueue) {
+                throw error;
+            }
+
+            // If we have caught a networking error from a deprecated api request, resolve it as unable to retry, otherwise the request will never resolve or reject.
+            request.resolve({jsonCode: CONST.JSON_CODE.UNABLE_TO_RETRY});
```

Comment:
@parasharrajat Would you please explain the downvote? Not every request has `resolve`. Only those that are not part of the sequential queue are ([source](https://github.com/Expensify/App/blob/472affc3e29eac5da7d456f4d3a5b617f294d8c1/src/libs/Network/index.js#L49-L51)). Although we have `isFromSequentialQueue` check just above, for the sake of consistency (and safety) we test `resolve` existence https://github.com/Expensify/App/blob/1e004aab1be7fdfe434a48d43c76b3b327d6ea09/src/libs/Middleware/Reauthentication.js#L104-L110 FWIW that logic was actually pulled from the Retry middleware https://github.com/Expensify/App/blob/472affc3e29eac5da7d456f4d3a5b617f294d8c1/src/libs/Middleware/Retry.js#L35-L37

Reply:
Yeah it's probably a good idea to be safe. However, I think it's also important to ask ourselves the following question. Is it currently possible for a request to be made from a different queue and for it to be missing the resolve function?

It looks like that could happen via [makeRequestWithSideEffects](https://github.com/Expensify/App/blob/e7cfb99aa2847b9cdf4467deb3a56ebe258c28cb/src/libs/API.js#L92). The request is not on the sequential queue and it doesn't have a resolve handler, so if it catches it will be a problem. Thanks for noticing that! I'll update it now.